common/server_protocolhandler.cpp

```diff
@@ -765,3 +751,24 @@ void Server_ProtocolHandler::resetIdleTimer()
     lastActionReceived = timeRunning;
     idleClientWarningSent = false;
 }
+
+bool Server_ProtocolHandler::isUserPermittedToJoinRoom(const QString roomPermission, const QString roomPrivilegeLevel)
+{
+    if (roomPermission.toLower() == "administrator" || roomPermission.toLower() == "moderator")
+        return false;
+
+    if (roomPermission.toLower() == "registered" && !(userInfo->user_level() & ServerInfo_User::IsRegistered))
+        return false;
+
+    if (roomPrivilegeLevel.toLower() != "none")
+    {
+        if (roomPrivilegeLevel.toLower() == "privileged")
```

Comment:
Can you rework this block so it's more defined? The else seems a little out of place w/o brackets for example